test/lit/basic/unit.wat

```diff
@@ -75,13 +75,6 @@
   (elem (i32.const 0) $z $big_negative $z $z $w $w $importedDoubles $w $z $cneg)
   ;; CHECK-TEXT:      (memory $0 4096 4096)
   ;; CHECK-BIN:      (memory $0 4096 4096)
-  ;; CHECK-BIN-NODEBUG:      (import "env" "_emscripten_asm_const_vi" (func $fimport$0 (type $1)))
-
-  ;; CHECK-BIN-NODEBUG:      (import "asm2wasm" "f64-to-int" (func $fimport$1 (type $5) (param f64) (result i32)))
-
-  ;; CHECK-BIN-NODEBUG:      (import "asm2wasm" "f64-rem" (func $fimport$2 (type $4) (param f64 f64) (result f64)))
-
-  ;; CHECK-BIN-NODEBUG:      (memory $0 4096 4096)
```

Comment:
Do you know why these move all the way down to line 1837? That makes things harder to read.

Reply:
This is an auto-update,  so no idea.  @tlively ?   Maybe this file was mistakenly manually updated at some point?

Reply:
It's because the name of the memory changed from `$0` to `$m0`. The auto update script matches input with output based on name, so when the name changes, the output moves to the bottom with the other unmatched output.

Reply:
Ah, that looks wrong, then - if the memory had a name `$0`, then it should stay `$0`?

Reply:
Sounds like a bug in the text parser maybe?

Reply:
Worked around the bug by naming the memory $m0 in the intput source.

Reply:
What's the bug?

Reply:
If I have a name in my wat file shouldn't than name be preserved rather than making up a new one.  In this case `$0` appears in the wat file, and but somebody decided to rename the memory to `$m0`.   IIUC names like `$m0` should only be injected when no other names was given.    In this case I would have thought the explicit `$0` would persist.

Reply:
That does seem like a bug, yeah. But I think the question is why was it noticed in this PR and not before. That is, all this PR should change is naming of memory imports from `mimport$N` to `$mN`, and that seems unrelated?

Reply:
This PR changes non-imported memories from `$N` -> $mM`, and leave imported memories as they are (`mimport$N`).

Reply:
Ah, right, it also renames non-imported ones. But I still don't see how such name changes can cause an existing name in the input to be forgotten.

Reply:
I think my change is just uncovering an existing bug.. my guess is that the memory name `$N` is perhaps not preserved in the name section since its assumed to be not worth preserving, or not a real name?   I'

Reply:
Isn't this WAI because the CHECK-BIN-NODEBUG run isn't preserving the name section?

Reply:
Sounds plausible.  My solution of naming the memory `$m0` to avoid the block moving down seems reasonable then I think.

Reply:
Hmm, sorry, I'm still confused - can you connect it back to the original question? I agree that CHECK-BIN-NODEBUG does not preserve the name section, but why does that fact explain this PR's original form (before the workaround) moving line 84 all the way down to 1837?

Reply:
I am guessing the renaming here somehow interacts with the auto-updater in a bad way, and my concern is that we can work around it here, but this PR modifies a great many tests, and maybe many of them will need such workarounds.

Reply:
Re-reading https://github.com/WebAssembly/binaryen/pull/6812#discussion_r1705871334 is seems like the auto-updater can only inline stuff when the names match.

In cases like this when we have no names section, I think the only way to make the names match is to explicitly choose the same names that that auto-namer would choose.  If you choose any other name then there won't be a match.  Is that right @tlively ?

Reply:
Right 👍

Reply:
To make sure I follow, so it worked before because the automatic name was `$N`, and now to keep that working we need to pick `$mN`?